libcontainerd/process_linux.go

```diff
@@ -79,7 +79,9 @@ func (r emptyReader) Read(b []byte) (int, error) {
 
 func openReaderFromFifo(fn string) io.Reader {
 	r, w := io.Pipe()
+	c := make(chan struct{})
 	go func() {
+		close(c)
 		stdoutf, err := os.OpenFile(fn, syscall.O_RDONLY, 0)
```

Comment:
Hmm, yeah, not sure if it buys you anything here. Worth to add  `// TODO` with little explanation.